bluesky/magics.py

```diff
@@ -107,55 +109,203 @@ def ct(self, line):
         self._ensure_idle()
         return None
 
-    positioners = []
+    _positioners = []
+    _detectors = []
+
+    @property
```

Comment:
Now that I look at this written out I realize it isn't going to work. Properties only work their magic on _instances_ of classes. We need `BlueskyMagics.positioners` to return `[]`, but in fact it would do this:

```py
In [16]: class BlueskyMagics():
    ...:     _positioners = []
    ...:     @property
    ...:     def positioners(self):
    ...:         print('hey')
    ...:         return self._positioners
    ...:     @positioners.setter
    ...:     def positioners(self, val):
    ...:         self._positioners = val
    ...:

In [17]: BlueskyMagics.positioners
Out[17]: <property at 0x10d524278>
```

The good news is (heh heh) we have to use metaclasses to make this work the way we want. But it works:

```
In [21]: class MetaclassForClassProperties(type):
    ...:     @property
    ...:     def positioners(self):
    ...:         print('hey')
    ...:         return self._positioners
    ...:     @positioners.setter
    ...:     def positioners(self, val):
    ...:         print('hey')
    ...:         self._positioners = val
    ...:

In [22]: class BlueskyMagics(metaclass=MetaclassForClassProperties):
    ...:     _positioners = []
    ...:
    ...:

In [23]: BlueskyMagics.positioners  # test access
hey
Out[23]: []

In [24]: BlueskyMagics.positioners.append('a')  # test mutatation
hey

In [25]: BlueskyMagics.positioners  # test access again
hey
Out[25]: ['a']

In [26]: BlueskyMagics.positioners = ['a', 'b']  # test setting
hey

In [27]: BlueskyMagics.positioners  # test access again
hey
Out[27]: ['a', 'b']
```